### PR TITLE
add order option for belongs_to field

### PR DESCRIPTION
**Pull request by MittchoBaroco**
add the posibility to order by one or more columns

- edit candidate_resources method
- add tests
- edit the doc

### CODE REVIEW

**Comment by MittchoBaroco** [comment]
updated version of #386 with tests and docs

**Comment by MittchoBaroco** [comment]
updated version of #386 with tests and docs

File: spec/lib/fields/belongs_to_spec.rb [hunk 1]
```diff
@@ -66,4 +66,19 @@
       end
     end
   end
+
+  describe "#resources" do
+    context "with `order` option" do
+      it "returns the resources in correct order" do
+        5.times { FactoryGirl.create(:customer) }
```
Comment (nickcharlton) [inline spec/lib/fields/belongs_to_spec.rb]: We could use `create_list` here, like: `create_list(:customer, 5)`, which will create 5 customers.

File: spec/lib/fields/belongs_to_spec.rb [hunk 2]
```diff
@@ -66,4 +66,19 @@
...
+        options = { order: "name" }
+        association = Administrate::Field::BelongsTo.with_options(options)
+        field = association.new(:customers, [], :view)
+
+        correct_order = Customer.all.order("name").map(&:id)
```
Comment (nickcharlton) [inline spec/lib/fields/belongs_to_spec.rb]: Could we use `pluck` here, instead of calling `map`?

File: spec/lib/fields/belongs_to_spec.rb [hunk 3]
```diff
@@ -66,4 +66,19 @@
...
+        resources = field.associated_resource_options.drop(1).map { |r| r[1] }
```
Comment (nickcharlton) [inline spec/lib/fields/belongs_to_spec.rb]: I wonder if this line is a little complex. Do you think we'd be able to simplify it a little?

Comment (MittchoBaroco) [inline spec/lib/fields/belongs_to_spec.rb]: I have updated it for `resources = field.associated_resource_options.compact.to_h.values` it's the simplest that I can think of.

the only way I found to test the data was to use the associated_resource_options method which returns an array like that: [nil, ["John doo", 1], ["master Yoda", 2}]

**Comment by MittchoBaroco** [comment]
updated 😄

File: spec/lib/fields/belongs_to_spec.rb [hunk 4]
```diff
@@ -66,4 +66,20 @@
...
+
+  describe "#resources" do
+    context "with `order` option" do
+      it "returns the resources in correct order" do
+        FactoryGirl.create_list(:customer, 5)
+        options = { order: "name" }
+        association = Administrate::Field::BelongsTo.with_options(options)
+        field = association.new(:customers, [], :view)
+
+        correct_order = Customer.all.order("name").pluck(:id)
```
Comment (nickcharlton) [inline spec/lib/fields/belongs_to_spec.rb]: We won't need to do `.all` here, as `.order` will be sufficient.

**Comment by MittchoBaroco** [comment]
@nickcharlton done 😄

**Comment by nickcharlton** [comment]
Great, thanks! I’m going to merge.

**Pull request merged**